### PR TITLE
Deprecated createTimeSeries()

### PR DESCRIPTION
**Pull request by vaibhavrdbhat**
Fixed problems for previous PR ([link](https://github.com/scalyr/Scalyr-Java-Client/pull/10)) on deprecating createTimeSeries(). Created new PR since I'm working on a separate branch now. Here is the old description:

Deprecated the createTimeSeries()/timeseriesID and added ability in timeseriesQuery() and TimeseriesQuerySpec to pass in function and filter on TimeseriesQuerySpec object.
Modified the tests as well to test this new functionality gets packed/unpacked correctly.

### CODE REVIEW

File: src/main/com/scalyr/api/query/QueryService.java [hunk 1]
```diff
@@ -386,22 +401,9 @@ private TimeseriesQueryResult unpackTimeseriesQueryResult(JSONObject rawApiRespo
...
-   * @param filter Specifies which log records to match, using the same syntax as the Expression field in the
-   *     query UI. To match all log records, pass null or an empty string.
-   * @param function Specifies the value to compute from the matching events. Has the same meaning as for the
-   *     numericQuery method.
-   *
-   * @throws ScalyrException if a low-level error occurs (e.g. network failure)
-   * @throws ScalyrServerException if the Scalyr service returns an error
+   * Deprecated. See git history for javadoc.
    */
+  @Deprecated
```
Comment (oliverhsu77) [inline src/main/com/scalyr/api/query/QueryService.java]: Annotations should be on the same line per Scalyr code style guide.

File: src/main/com/scalyr/api/query/QueryService.java [hunk 2]
```diff
@@ -345,9 +350,8 @@ public TimeseriesQueryResult timeseriesQuery(TimeseriesQuerySpec[] queries)
    * Parameters for a single timeseries query.
    */
   public static class TimeseriesQuerySpec {
-    /**
-     * ID of the timeseries to query, as returned by a previously call to createTimeseries.
-     */
+    /** This field is deprecated. Use 'filter' and 'function' instead. */
+    @Deprecated
```
Comment (oliverhsu77) [inline src/main/com/scalyr/api/query/QueryService.java]: Annotations should be on the same line per Scalyr code style guide.

File: src/main/com/scalyr/api/query/QueryService.java [hunk 3]
```diff
@@ -327,10 +327,15 @@ public TimeseriesQueryResult timeseriesQuery(TimeseriesQuerySpec[] queries)
...
+      if (query.timeseriesId != null) { // Existence of a timeseriesID takes precedence, although it will be deprecated eventually.
+        queryJson.put("timeseriesId", query.timeseriesId);
+      } else { // If using new API (no timeseriesID), use filter and function.
+        if (query.filter   != null) { queryJson.put("filter", query.filter);     }
+        if (query.function != null) { queryJson.put("function", query.function); }
+      }
       queryJson.put("startTime", query.startTime);
-      queryJson.put("endTime", query.endTime);
-      queryJson.put("buckets", query.buckets);
+      queryJson.put("endTime"  , query.endTime  );
```
Comment (oliverhsu77) [inline src/main/com/scalyr/api/query/QueryService.java]: Please restore original formatting, which was correct.

**Comment by oliverhsu77** [comment]
Done with review.  Few style comments.

**Comment by vaibhavrdbhat** [comment]
Done

**Pull request merged**